backend/application/syson-sysml-import/src/main/java/org/eclipse/syson/sysml/ASTTransformer.java

```diff
@@ -12,171 +12,62 @@
-            new MapperArray(this.objectFinder, this.mappingStates),
-            new MapperCreateKnownType(this.objectFinder, this.mappingStates),
-            new MapperGenericAttributes(this.objectFinder, this.mappingStates),
-            new MapperGenericReferences(this.objectFinder, this.mappingStates),
-            new MapperFeatureTyping(this.objectFinder, this.mappingStates),
-            new MapperFlowConnectionUsage(this.objectFinder, this.mappingStates),
-            new MapperMembership(this.objectFinder, this.mappingStates),
-            new MapperMembershipImport(this.objectFinder, this.mappingStates),
-            new MapperNamespaceImport(this.objectFinder, this.mappingStates),
-            new MapperRedefinition(this.objectFinder, this.mappingStates),
-            new MapperSubclassification(this.objectFinder, this.mappingStates),
-            new MapperFeatureChaining(this.objectFinder, this.mappingStates),
-            new MapperSubsetting(this.objectFinder, this.mappingStates),
-            new MapperUsage(this.objectFinder, this.mappingStates),
-            new MapperOperatorExpression(this.objectFinder, this.mappingStates),
-            new MapperLiteralInteger(this.objectFinder, this.mappingStates),
-            new MapperLiteralRational(this.objectFinder, this.mappingStates),
-            new MapperFeatureReferenceExpression(this.objectFinder, this.mappingStates),
-            new MapperConnectionUsage(this.objectFinder, this.mappingStates),
-            new MapperEventOccurrenceUsage(this.objectFinder, this.mappingStates),
-            new MapperDependency(this.objectFinder, this.mappingStates),
-            new MapperSpecialization(this.objectFinder, this.mappingStates),
-            new MapperMembershipReference(this.objectFinder, this.mappingStates),
-            new MapperLiteralString(this.objectFinder, this.mappingStates),
-            new MapperReferenceSubsetting(this.objectFinder, this.mappingStates),
-            new MapperConjugatedPortTyping(this.objectFinder, this.mappingStates),
-            new MapperComment(this.objectFinder, this.mappingStates)
-            );
-
     private final Logger logger = LoggerFactory.getLogger(ASTTransformer.class);
 
-    public Resource convertResource(InputStream input, List<EObject> list) {
-        list.parallelStream().forEach((t) -> {
-            if (t != null) {
-                this.objectFinder.putEObject(t);
+    public Resource convertResource(InputStream input, ResourceSet editingDomainResourceSet) {
+
+        Resource result = new JSONResourceFactory().createResource(new JSONResourceFactory().createResourceURI(null));
+        editingDomainResourceSet.getResources().add(result);
+        if (input != null) {
+            JsonNode astJson = readAst(input);
+            if (astJson != null) {
+                logger.info("Create the Root eObject containment structure");
+                List<EObject> rootSysmlObjects = AstTreeParser.parseAst(astJson);
```

Comment:
Why do you have/call a static method here?
Unless I'm missing something, there is no added value to have a static access here, this should be replace by an instantiation of `AstTreeParser` and a call to `parseAst`.

Reply:
OK